curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md

```diff
@@ -9,17 +9,21 @@ challengeType: 2
 As a reminder, this project is being built upon the following starter project on <a href='https://glitch.com/#!/import/github/freeCodeCamp/boilerplate-advancednode/'>Glitch</a>, or cloned from <a href='https://github.com/freeCodeCamp/boilerplate-advancednode/'>GitHub</a>.
 As in, any user can just go to /profile whether they authenticated or not by typing in the url. We want to prevent this by checking if the user is authenticated first before rendering the profile page. This is the perfect example of when to create a middleware.
 The challenge here is creating the middleware function <code>ensureAuthenticated(req, res, next)</code>, which will check if a user is authenticated by calling passports isAuthenticated on the <em>request</em> which in turn checks for <em>req.user</em> is to be defined. If it is then <em>next()</em> should be called, otherwise we can just respond to the request with a redirect to our homepage to login. An implementation of this middleware is:
-<pre>function ensureAuthenticated(req, res, next) {
-  if (req.isAuthenticated()) {
-      return next();
-  }
-  res.redirect('/');
-};</pre>
+<blockquote>
+function ensureAuthenticated(req, res, next) {<br>
+&nbsp;&nbsp;if (req.isAuthenticated()) {<br>
+&nbsp;&nbsp;&nbsp;&nbsp;return next();<br>
+&nbsp;&nbsp;}<br>
+&nbsp;&nbsp; res.redirect('/');<br>
+};
+</blockquote>
 Now add <em>ensureAuthenticated</em> as a middleware to the request for the profile page before the argument to the get request containing the function that renders the page.
-<pre>app.route('/profile')
-  .get(ensureAuthenticated, (req,res) => {
-       res.render(process.cwd() + '/views/pug/profile');
-  });</pre>
+<blockquote>
+app.route('/profile')<br>
```

Comment:
I also debate whether or not to indent the method (chained or not).  Would either of the following be more readable?
```
app.route('/profile').get(
  ensureAuthenticated,
  (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);

```
OR not indenting `.get`
```
app.route('/profile')
.get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);
```

Reply:
I guess not indenting looks better

Reply:
Rather, how doeds this look:
```
app
  .route('/profile')
  .get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);

Reply:
@thecodingaviator Go with the last one above.

Reply:
The one I posted or the latter one you said, I’m sorry a bit confused

Reply:
```
app
  .route('/profile')
  .get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);
```
Your version.